src/runtime/tests/refiner-test.ts

```diff
@@ -338,3 +338,139 @@ describe('SQLExtracter', () => {
     assert.strictEqual(query, 'SELECT * FROM table;');
   });
 });
+
+describe('Polynomial', () => {
+  it('tests coeffs getters, setters and degree works', () => {
+      let pn = new Polynomial([0, 1, 2]);       // 2a^2 + a
+      assert.deepEqual(pn.coeffs, [0, 1, 2]);
+      assert.strictEqual(pn.degree(), 2);
+      pn = new Polynomial([0, 1, 2, 0, 0, 0]);  // 2a^2 + a
+      assert.strictEqual(pn.degree(), 2);
+      pn = new Polynomial([0, 0, 0, 0]);        // 0
+      assert.deepEqual(pn.coeffs, [0]);
+      assert.strictEqual(pn.degree(), 0);
+      pn = new Polynomial([]);                  // 0
+      assert.deepEqual(pn.coeffs, [0]);
+      assert.strictEqual(pn.degree(), 0);
+      pn = new Polynomial();                    // 0
+      assert.deepEqual(pn.coeffs, [0]);
+      assert.strictEqual(pn.degree(), 0);
+  });
+  it('tests polynomial addition works', () => {
+    let pn1 = new Polynomial([0, 1, 2]);        // 2a^2 + a
+    let pn2 = new Polynomial();                 // 0
+    let sum = Polynomial.add(pn1, pn2);         // 2a^2 + a
+    assert.deepEqual(sum.coeffs, [0, 1, 2]);
+    assert.strictEqual(sum.degree(), 2);
+    pn1 = new Polynomial([0, 1, 2]);            // 2a^2 + a
+    pn2 = new Polynomial([3, 2, 1]);            // a^2 + 2a + 3
+    sum = Polynomial.add(pn1, pn2);             // 3a^2 + 3a + 3
+    assert.deepEqual(sum.coeffs, [3, 3, 3]);
+    assert.strictEqual(sum.degree(), 2);
+    pn1 = new Polynomial([0, 1, 2]);            // 2a^2 + a
+    pn2 = new Polynomial([3, 0, 0, 0, 0]);      // 3
+    sum = Polynomial.add(pn1, pn2);             // 2a^2 + a + 3
+    assert.deepEqual(sum.coeffs, [3, 1, 2]);
+    assert.strictEqual(sum.degree(), 2);
+  });
+  it('tests polynomial negation works', () => {
+    const pn1 = new Polynomial([0, 1, 2]);      // 2a^2 + a
+    const neg = Polynomial.negate(pn1);         // -2a^2 -a
+    // TODO(ragdev): Why does 0 fail in the assertion below but -0 works?
```

Comment:
Had a little look, seems that 0 and -0 are not deep equal for mocha's `deepEqual` purposes.
This is somewhat concerning, especially as `0==-0` and `0===-0` are both true.

```
  it.only('maths', async () => {
    assert.equal(0, -0);
    assert.deepEqual(0, 0-0);
    assert.deepEqual(0, 0+(-0));
    assert(0 == -0);
    assert(0 === -0);
    assert.deepEqual(0, -0); // This fails.
  });
```

Reply:
Turns out this is expected behavior https://tc39.es/ecma262/#sec-samevalue
and our assertion framework's `deepEqual` is using a very strict version of equality, even more strict that `===` (strict equality).

It seems the decision may be due to handling 1/0 and 1/-0 as + and - infinity.